### DISCUSSION TITLE
V3 suggestions by Joe W.

### DISCUSSION THREAD

**Discussion by zFernand0** [zowe/zowe-explorer-vscode#2416] [Ideas]
During yesterday's (August 8th, 2023) Architecture call, Joe W. shared some interesting ideas for future enhancement in Zowe Explorer. Here is a summary of the ideas:
- Default job search filters 
  - We could add quick-pick items for filtering by status: active, input, or output
  - These new items could be added right below `Search by Job ID`
  - ![image](https://github.com/zowe/vscode-extension-for-zowe/assets/37381190/7fc670eb-40dc-43d7-b3ee-f9d336aaacf5)
  - Note: `Held` is not supported by z/OSMF so we have to do that one on the client
- Default syslog favorites for each added profile
  - This seems to be a common enough operation that we could add as a default favorite for any visible (not-hidden) profile.
  - ![image](https://github.com/zowe/vscode-extension-for-zowe/assets/37381190/3ed9f506-c79c-4c8e-96e0-0149ffbb5b3a)
- Open USS files and datasets in read-only mode if they are actively being edited in ISPF 
  - The user get an Enqueue error when they try to save a file that it currently locked by another editor
  - ![image](https://github.com/zowe/vscode-extension-for-zowe/assets/37381190/184eafc8-c4ba-49e8-8faf-9faa8479128a)
- Toggle HEX mode on DS and USS 
  - Trae suggested to use the toolbar, located at the top right corner of your editor instead of a new context (right-click) menu item
  - ![image](https://github.com/zowe/vscode-extension-for-zowe/assets/37381190/0b8e3856-80d2-47ae-95e1-2c1ecc65174a)
- Increase search history
  -  We should consider making it customizable
  - ![image](https://github.com/zowe/vscode-extension-for-zowe/assets/37381190/90287b1c-521d-44d8-b345-f9057ed3e22e)
- Have search history per profile vs per tree
  - It's possible that some (if not all) search history only applies to the profile/lpar where you performed the search.
  - ![image](https://github.com/zowe/vscode-extension-for-zowe/assets/37381190/71b56dbd-0776-4202-9eeb-920a224b7847)

Additional ideas that the team discussed:
- How to backup Search History (now that ZE is moving to Local Storage)?
  - Before: 
![image](https://github.com/zowe/vscode-extension-for-zowe/assets/37381190/0f829770-f904-4b2e-aae9-baad4d777dba)
  - After:
  ![image](https://github.com/zowe/vscode-extension-for-zowe/assets/37381190/345d839f-eb3a-42f9-a7c1-cbd9c0c7f270)

**Comment by zFernand0** [comment]
@Joe-Winchester 
Please let me know if I missed anything from the Architecture call.
Also, feel free to add any new suggestions here 😋

**Reply by Joe-Winchester** [reply]
Hi Fernando,  sorry for the late reply.  Got behind with e-mails so just catching up now.
Everything looks great and thank you for taking notes ! 
For the locked files I think it's difficult to know ahead of time whether there is an ENQUE on the data set so it's going to be tough to pre-emptively open it in read only mode, however if on save the error message about the 500, rc6, .... occurs maybe it'd be possible to parse and detect this, and then allow some kind of comparison to occur to allow the user to save ?  Or maybe there is a way to detect which userID has the lock (which I think might be in the REST response). 
Another thing that might be interesting alongside the SYSLOG option (which is equivalent to the LOG menu action in SDSF) is to have a console option also.  This could lean down on the console SDK that underpins the `zowe console` CLI.  However for SYSLOG and console the volume of data likely could be large, so there might to be a way to throttle the data being retrieved.
